grpc/src/main/java/com/linecorp/armeria/internal/client/grpc/ArmeriaClientCall.java

```diff
@@ -417,17 +431,17 @@ public void onNext(DeframedMessage message) {
                 GrpcUnsafeBufferUtil.storeBuffer(buf, msg, ctx);
             }
 
-            try (SafeCloseable ignored = ctx.push()) {
-                assert listener != null;
-                listener.onMessage(msg);
-            }
+            contextAwareExecutor.execute(() -> {
+                try {
+                    assert listener != null;
+                    listener.onMessage(msg);
+                } catch (Throwable t) {
+                    close(GrpcStatus.fromThrowable(t), new Metadata());
+                }
+            });
         } catch (Throwable t) {
-            final Status status = GrpcStatus.fromThrowable(t);
-            req.close(status.asRuntimeException());
```

Comment:
This is a slight behavior change. Before `req.close(t)`; after: `req.abort(t)` (happening as part of `close(status, metadata)`. Is there any material difference between the two?